legateboost/utils.py

```diff
@@ -116,7 +116,7 @@ def set_col_by_idx(a: cn.ndarray, b: cn.ndarray, delta: float) -> None:
     bools = b[:, cn.newaxis] == range[cn.newaxis, :]
     a -= a * bools
     a += delta * bools
-    return
+    return a
```

Comment:
Confusing that it is in-place but not in-place, but seems good, thanks for tracking this down!

TBH this smells like another cupynumeric issue to me that fell out of more agressively deferring to NumPy.  (Admittedly, because NumPy is horrible with 0-D arrays and I never got around to fix it with all the potential downstream fallout).
The input seems clearly an array, yet the in-place operator isn't in-place.